src/verifier/ReachabilitySearcher.cpp

```diff
@@ -107,25 +107,34 @@ void ReachabilitySearcher::PrintResults(const std::vector<QueryResultPair>& resu
         spdlog::info("===================="); // Delimiter to make it easier to read
         spdlog::info("{0} : {1}", ConvertASTToString(*r.query), r.answer);
         auto stateHash = r.acceptingStateHash;
+        auto state = r.acceptingState;
         std::vector<std::string> trace{};
         while(stateHash != 0) { // 0 indicates "no parent"
             spdlog::trace("trace hash: {0}", stateHash);
-            auto prevState = Passed.find(stateHash);
-            if(prevState != Passed.end()) {
-                trace.push_back(prevState->second.tta.GetCurrentStateString());
-                if(stateHash == Passed[stateHash].prevStateHash) {
-                    spdlog::critical("Breaking out of infinite loop. Please make sure that the trace is correct");
-                    break;
+            auto exists = Passed.find(stateHash) != Passed.end();
+            if(exists) {
+                auto range = Passed.equal_range(stateHash);
+                auto count = Passed.count(stateHash);
+                if(count > 1)
+                    spdlog::warn("HASH COLLISIONS: {0}", count);
+
+                stateHash = range.first->second.prevStateHash;
```

Comment:
NOTE: This does not handle the collision, it just simply picks the first one! - If there are hash collisions in your results, the trace cannot be trusted!